### PR TITLE
Add specs for PagedEnumerable

### LINKED ISSUE #217: Add tests for PagedEnumerable
There are no tests for it currently.

### PR DESCRIPTION
**Pull request by blowmage**
[closes #217]

### CODE REVIEW

File: spec/google/gax/paged_enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    paged_enum.each do |user|
+      exp_names << user.name
+    end
+    expect(exp_names).to eq(%w[foo bar baz bif])
+  end
+
+  it 'enumerates all pages' do
+    api_responses = [
+      Google::Protobuf::GoodPagedResponse.new(
+        users:           [
```
Comment (dazuma) [inline spec/google/gax/paged_enumerable_spec.rb]: what's up with the whitespace?

**Review by dazuma — APPROVED** [review]

File: spec/google/gax/paged_enumerable_spec.rb (hunk 1 above)
Comment (blowmage) [reply spec/google/gax/paged_enumerable_spec.rb]: I legit don't know. I ran `rubocop --auto-correct` and assume it made it. Removed.

**Comment by blowmage** [comment]
You add 125 lines of tests and code coverage still decreases by 0.01%...

**Comment by dazuma** [comment]
Thanks a lot codecov. WTF are you doing anyway?

**Pull request merged**